# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/OffThreadScriptCompilation.h
# Commit: 2bc5949902ac
# Full Hash: 2bc5949902acc042205fc7e85260bbbc37dbedc7
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2021-09-18 09:46:57
# Regressor Bug: 1688788
# File Overlap Count: 1
# Description:
#   Bug 1688788 - Convert ScriptPreloader to manage Stencils, not Scripts. r=kmag,tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113013
# ==============================================================================

diff -r e660e2c3c3d1 -r 2bc5949902ac js/public/OffThreadScriptCompilation.h
--- a/js/public/OffThreadScriptCompilation.h	Sat Sep 18 01:29:45 2021 +0000
+++ b/js/public/OffThreadScriptCompilation.h	Sat Sep 18 01:29:45 2021 +0000
@@ -18,16 +18,15 @@
 
 #include "jstypes.h"  // JS_PUBLIC_API
 
-#include "js/GCVector.h"  // JS::GCVector
+#include "js/CompileOptions.h"  // JS::ReadOnlyCompileOptions
+#include "js/GCVector.h"        // JS::GCVector
+#include "js/Transcoding.h"     // JS::TranscodeSource
 
 struct JS_PUBLIC_API JSContext;
 class JS_PUBLIC_API JSScript;
 
 namespace JS {
 
-class JS_PUBLIC_API ReadOnlyCompileOptions;
-struct TranscodeSource;
-
 template <typename UnitT>
 class SourceText;
 
@@ -156,16 +155,6 @@
 extern JS_PUBLIC_API void CancelOffThreadScriptDecoder(JSContext* cx,
                                                        OffThreadToken* token);
 
-// Decode multiple JSScript from the sources.
-extern JS_PUBLIC_API OffThreadToken* DecodeMultiOffThreadScripts(
-    JSContext* cx, const ReadOnlyCompileOptions& options,
-    mozilla::Vector<TranscodeSource>& sources,
-    OffThreadCompileCallback callback, void* callbackData);
-
-extern JS_PUBLIC_API bool FinishMultiOffThreadScriptsDecoder(
-    JSContext* cx, OffThreadToken* token,
-    MutableHandle<GCVector<JSScript*>> scripts);
-
 extern JS_PUBLIC_API void CancelMultiOffThreadScriptsDecoder(
     JSContext* cx, OffThreadToken* token);
 